server/src/test/java/org/elasticsearch/common/RoundingTests.java

```diff
@@ -1010,6 +1010,12 @@ public void testIntervalTwoTransitions() {
         assertThat(rounding.round(time("1982-11-10T02:51:22.662Z")), isDate(time("1982-03-23T05:00:00Z"), tz));
     }
 
+    public void testHugeTimeInterval() {
+        ZoneId tz = ZoneId.of("Asia/Tehran");
+        Rounding rounding = Rounding.builder(TimeValue.timeValueDays(80000)).timeZone(tz).build();
+        assertThat(rounding.round(time("2078-11-10T02:51:22.662Z")), isDate(time("1970-01-01T00:00:00+03:30"), tz));
+    }
```

Comment:
I couldn't write a test locally that'd fail without going to pretty intense lengths. So I just used the worse case I could find.